generators/client/templates/angular/src/main/webapp/app/account/password-reset/finish/password-reset-finish.component.html.ejs

```diff
@@ -43,50 +43,50 @@
             </div>
 
             <div *ngIf="!keyMissing">
-                <form *ngIf="!success" name="form" role="form" (ngSubmit)="finishReset()" #passwordForm="ngForm">
+                <form *ngIf="!success" name="form" role="form" (ngSubmit)="finishReset()" [formGroup]="passwordForm">
                     <div class="form-group">
                         <label class="form-control-label" for="password" jhiTranslate="global.form.newpassword">New password</label>
-                        <input type="password" class="form-control" id="password" name="password" #passwordInput="ngModel"
+                        <input type="password" class="form-control" id="password" name="password"
                                placeholder="{{'global.form.newpassword.placeholder' | translate}}"
-                               [(ngModel)]="resetAccount.password" minlength=4 maxlength=50 required>
-                        <div *ngIf="passwordInput.dirty && passwordInput.invalid">
+                               formControlName="newPassword">
+                        <div *ngIf="passwordForm.get(['newPassword']).invalid && (passwordForm.get(['newPassword']).dirty || passwordForm.get(['newPassword']).touched)">
```

Comment:
I generally use a helper function like `isFieldInvalid(field: string)` to reduce the complexity within the HTML File.
WDYT about doing this for the error checks?

Reply:
If used at more than one place, I generally create getter methods and access those in template and component class like below:

```
get password(): FormControl {
  return this.passwordForm.get('newPassword');
}
```

// use in template like:
`password.invalid `

Reply:
Not a bad idea but I would prefer a wrapper to check invalid, dirty and touched and then return a boolean

Reply:
wrapper can also leverage getter() functions. Point is to reduce complexity from templates (at moment, I see `passwordForm.get(['newPassword'])` code is repeated at more than 5+ places in this template and any future change in property name or object structure would require modification at all these places.

Reply:
You're right, but I would do a wrapper which takes in a parameter `fieldName` (like `newPassword`) and returns a boolean.
With getters we couldn't do that way, we would need to do a `isInvalid` method per field, if I understood you correctly

Reply:
I'm not fan of wrapper: imagine a MOA want to show that some fields are mandatory (errored) without the end user to enter on the field.
If it's a sine qua non feature I'll do, but I would personally prefer without it

Reply:
Ok, let's keep it that way, it is indeed more flexible